src/lib/components/TocNav.svelte

```diff
@@ -12,94 +13,194 @@
     } = getTocCtx();
 
     $: progress = Math.max(...$activeHeadingIdxs) / ($headingsTree.length - 1);
+
+    function slideFade(
+        node: HTMLElement,
+        {
+            delay = 0,
+            duration = 400,
+            easing = cubicOut
+        }: { delay?: number; duration?: number; easing?: (t: number) => number } = {}
+    ) {
+        const initialHeight = node.offsetHeight;
+
+        return {
+            delay,
+            duration,
+            easing,
+            css: (t: number) => {
+                return `
+                    opacity: ${t};
+                    height: ${t * initialHeight}px;
+                    overflow: hidden;
+                `;
+            }
+        };
+    }
```

Comment:
Can we also do this animation with CSS? I think that would better fit the direction we're moving in. Wdyt @thejessewinton?

Reply:
Not sure if we probably can (due to `t * initialHeight`, but correct me if I am wrong), but I think I've seen a similar use-case somewhere around the repo. Lmk what we decide.

Reply:
ping @ernstmul @thejessewinton.